packages/snackbar/index.tsx

```diff
@@ -0,0 +1,167 @@
+//
+// Copyright (c) 2019 Google, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as React from 'react';
+import classnames from 'classnames';
+import {IMDCSnackbarAdapter, IMDCSnackbarFoundation} from './types';
+
+// @ts-ignore no .d.ts file
+import {MDCSnackbarFoundation} from '@material/snackbar';
+
+type Props = {
+  message: string,
+  className?: string,
+  timeoutMs?: number,
+  closeOnEscape?: boolean,
+  actionText?: string,
+  leading?: boolean,
+  stacked?: boolean,
+  onBeforeOpen?: () => void,
+  onOpen?: () => void,
+  onBeforeClose?: (reason: string) => void,
+  onClose?: (reason: string) => void,
+  onAnnounce?: () => void,
+};
+
+type State = {
+  classes: Set<string>,
+};
+
+export class Snackbar extends React.Component<Props, State> {
+  foundation: IMDCSnackbarFoundation
+  constructor(props: Props) {
+    super(props);
+    const {timeoutMs, closeOnEscape, leading, stacked} = this.props;
+    const classes = new Set();
+    if (leading) {
+      classes.add('mdc-snackbar--leading');
+    }
+
+    if (stacked) {
+      classes.add('mdc-snackbar--stacked');
+    }
+
+    this.state = {
+      classes,
+    };
+
+    this.foundation = new MDCSnackbarFoundation(this.adapter);
+    if (timeoutMs) {
+      this.foundation.setTimeoutMs(timeoutMs);
+    }
+
+    if (closeOnEscape) {
+      this.foundation.setCloseOnEscape(closeOnEscape);
+    }
+  }
+  get adapter(): IMDCSnackbarAdapter {
+    return {
+      addClass: (className: string) => {
+        const {classes} = this.state;
+        classes.add(className);
+        this.setState({
+          classes,
+        });
+      },
+      removeClass: (className: string) => {
+        const {classes} = this.state;
+        classes.delete(className);
+        this.setState({
+          classes,
+        });
+      },
+      announce: () => {
+        // Usually it works automatically if this component uses conditional rendering
```

Comment:
What does this comment mean?

Reply:
In MDC Snackbar component there are some hacks to make `announce` work correctly.
With React, it works correctly out of the box. As soon as it renders initially, screenreader announces the text